mteb/abstasks/AbsTaskInstructionRetrieval.py

```diff
@@ -0,0 +1,312 @@
+
+                newly_irrelevant_qrels = self.create_qrel_diff(self.og_relevant_docs[lang][split], self.changed_relevant_docs[lang][split])
+                changed_scores[lang] = utils.evaluate_change(results_og[lang], results_changed[lang], newly_irrelevant_qrels)
+
+                changed_scores[lang]["individual"] = {
+                    "original": scores_og[lang],
+                    "changed": scores_changed[lang],
+                    "base": scores_base[lang]
+                }
+        else:
+            corpus, queries, og_relevant_docs, changed_relevant_docs = self.corpus[split], self.queries[split], self.og_relevant_docs[split], self.changed_relevant_docs[split]
+            og_instructions, changed_instructions = self.og_instructions[split], self.changed_instructions[split]
+            top_ranked = self.top_ranked[split]
+            keywords, short_instructions = self.keywords[split], self.short_instructions[split]
+            scores_og, results_og = self._evaluate_monolingual(retriever, corpus, queries, og_relevant_docs, "og", og_instructions, top_ranked, None, **kwargs)
+            scores_changed, results_changed = self._evaluate_monolingual(retriever, corpus, queries, changed_relevant_docs, "changed", changed_instructions, top_ranked, None, **kwargs)
+            scores_base, results_base = self._evaluate_monolingual(retriever, corpus, queries, og_relevant_docs, "base", defaultdict(str), top_ranked, None, **kwargs)
+
+            newly_irrelevant_qrels = self.create_qrel_diff(self.og_relevant_docs[split], self.changed_relevant_docs[split])
+            changed_scores = utils.evaluate_change(results_og, results_changed, newly_irrelevant_qrels)            
+
+            changed_scores["individual"] = {
+                "original": scores_og,
+                "changed": scores_changed,
+                "base": scores_base
+            }
+
+            if self.do_length_ablation:
+                scores_w_keywords = self._evaluate_monolingual(retriever, corpus, queries, og_relevant_docs, "keywords", keywords, top_ranked, None, **kwargs)
+                scores_w_short_instr = self._evaluate_monolingual(retriever, corpus, queries, og_relevant_docs, "short_instructions", short_instructions, top_ranked, None, **kwargs)
+                changed_scores["length_ablation"] = {
+                    "keywords": scores_w_keywords,
+                    "short_instructions": scores_w_short_instr
+                }
+
+        return changed_scores
+
+    def _evaluate_monolingual(self, retriever, corpus, queries, relevant_docs, qrels_name: str, instructions, top_ranked, lang=None, **kwargs):
+        start_time = time()
+
+        # do the results by query and relevant docs only
+        all_results = []
+        for query_id in tqdm.tqdm(list(queries.keys()), leave=True):
```

Comment:
Unlike standard retrieval, this is just re-ranking the top N, so we do it query by query instead of all at once